src/data/set/lattice.lean

```diff
@@ -358,6 +358,14 @@ theorem bUnion_union (s t : set α) (u : α → set β) :
   (⋃ x ∈ s ∪ t, u x) = (⋃ x ∈ s, u x) ∪ (⋃ x ∈ t, u x) :=
 supr_union
 
+@[simp] lemma bUnion_of_singleton_of_coe (s : set α) :
+  (⋃ (i : s), {i} : set α) = s :=
```

Comment:
I think this name is incorrect - `⋃ (i ∈ s)` is a `bUnion`, while `⋃ (i : s)` is just a `Union`. That probably indicates that this lemma belongs at a different point in the file too.

Reply:
I am only moderately aware of these subtleties, so I may be wrong about this.  In the file, there are a lot of lemmas where there is a separate Type, the Union ranges over the separate Type, and there is an explicit function from this other Type to set \a.  In this case, probably the separate Type would be s (viewed as a Type) and the function is the "singleton of coercion".  I am not sure that I can formulate a simpler lemma, though.

Do you have any suggestion?  Also, I browsed the file, but I could not find where this lemma would go, if not where I put it.

Reply:
I think this lemma belongs right next to `Union_of_singleton`, and called `Union_of_singleton_coe`.

https://github.com/leanprover-community/mathlib/blob/06c452762dbdc4ef9fad23b4d6e6fa9ae481e3fe/src/data/set/lattice.lean#L545-L546

I'm not suggesting you change the lemma, but just that the word `bUnion` should not appear in its name.

Reply:
I had missed your reply: I just moved the lemma to where you suggest and renamed it!  Thanks!